partiql-ast/src/main/resources/partiql_ast.ion

```diff
@@ -489,22 +489,10 @@ expr::[
     type:   '.set_op',
     lhs:    expr,
     rhs:    expr,
-    outer:  optional::bool,
   },
 
-  // The PartiQL `<sfw>` query expression, think SQL `<query specification>`
-  s_f_w::{
-    select:   select,              // oneof SELECT / SELECT VALUE / PIVOT
-    exclude:  optional::exclude,
-    from:     from,
-    let:      optional::let,
-    where:    optional::expr,
-    group_by: optional::group_by,
-    having:   optional::expr,
-    set_op:   optional::{
-      type:    '.set_op',
-      operand: '.expr.s_f_w',
-    },
+  query_set::{
```

Comment:
(self-review) primary addition of this PR is adding `query_set` and `query_expr` to the AST. This modeling allows us to define `ORDER BY`, `LIMIT`, and `OFFSET` on SFW queries as well as SQL set ops.

Modeling is partially based off
- partiql-lang-rust -- https://github.com/partiql/partiql-lang-rust/blob/e5b8ce7497f515bc998a31cd8b45fb2f3e9287a2/partiql-ast/src/ast.rs#L238-L244
- similar modeling in the sqlparser Rust crate -- https://docs.rs/sqlparser/0.48.0/sqlparser/ast/struct.Query.html

Reply:
Nice work. Would we want to add the naming followed by PLR and SQLParser? AKA `Query` is the top-level product type holding the expression, order-by, etc. And the union types `QuerySet` (PLR) or `SetExpr` (SQLParser) holding the different variants (expr, select, wrapped Query, values, etc).

Especially since the Java name will now be: `Expr.QuerySet` and `QueryExpr`. I found it a bit confusing in the rest of the code. Could just be: `Expr.Query` and `Expr.Query.Body`.

Reply:
I added some more context on PLR's AST definition in the partiql-tests PR https://github.com/partiql/partiql-tests/pull/121#discussion_r1719123490. The AST definition I used in this PR allows for us to be more strict in what expressions we want to support with `LIMIT`, `OFFSET`, and `ORDER BY`. For now, I'm thinking we only permit those clauses on
- SFW
- SQL set ops
- PartiQL bag ops

PLR also allows those clauses on general expressions. Since it's not in the PartiQL spec, I haven't chosen to allow those clauses on general expressions.

I agree that the previous naming was confusing, so updated to use `query_body` rather than `query_expr`.